backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -79,19 +80,39 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 			return shopIdToListOfTokensMap.get(shopId).stream().filter(token -> token.getStatus() == Status.ACTIVE)
 					.collect(Collectors.toList());
 		}
-
-		// TODO(b/158975796): Simplify logging across all backend APIs
+    
+		// TODO(b/158975796): Simplify logging across all backend APIs.
 		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
 				"Tried to fetch tokens of a shop which is not in the ACTIVE state.");
 		return new ArrayList<>();
 	}
 
-	// Method returns all shops keeping in mind the feature of restoring shops later on.
+	public Token getNewToken(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			Integer newTokenNumber = shopIdToLastAllotedTokenMap.get(shopId).incrementAndGet();
+			UUID uuid = UUID.randomUUID();
+			Token newToken = new Token(uuid, shopId, newTokenNumber);
+			newToken.setStatus(Token.Status.ACTIVE);
+			tokenMap.put(uuid, newToken);
+			shopIdToListOfTokensMap.get(shopId).add(newToken);
+			shopIdToListOfTokensMap.put(shopId, shopIdToListOfTokensMap.get(shopId));
```

Comment:
Do we need this line now?

Reply:
yes removed it. Should have removed it after the call. Sorry